extension/js/common/core/attachment.ts

```diff
@@ -204,7 +204,7 @@ export class Attachment {
     } else if (this.name === 'message' && isBodyEmpty) {
       // treat message as encryptedMsg when empty body for the 'message' attachment
       return 'encryptedMsg';
-    } else if (this.name.match(/(\.pgp$)|(\.gpg$)|(\.[a-zA-Z0-9]{3,4}\.asc$)/g)) {
+    } else if (this.name.match(/(\.pgp$)|(\.gpg$)|(\.[a-zA-Z0-9]{3,4}\.asc$)/g) && !this.isPublicKey() && !this.isPrivateKey()) {
```

Comment:
what about adding test for this case? probably shouldn't be too complicated, but will help to not break this solution in the future

Reply:
Thank you for the remarks. You're right, and I missed that opportunity as I initially thought it wouldn't need testing. I'll definitely need to add a test for this.